### PR TITLE
Solr 8.4

### LINKED ISSUE #187: Update catalog's CJKFoldingFilterFactory
It looks like the one we're using is 30-something commits behind stanford's master. Do we want to update?

@tampakis says they've replaced a lot of their bigram tokenization with dictionary-based tokenization.

### LINKED ISSUE #188: Fix CJK tokenization beyond stop-gap
Currently we've disabled bigrams for CJK queries, which we expect will decrease the quality of the keyword match retrieval behavior.

https://github.com/pulibrary/pul_solr/issues/187 might help.

Discussion on blacklight slack channel suggests that upgrading to solr 8.1 would mitigate the bug; however the queries that previously caused the crash would fail.

To patch the bug without upgrading solr we could use https://issues.apache.org/jira/browse/SOLR-13336. Note M. Gibney endorses a different patch, but using it would mean departing from solr core longer-term:

"I would expect that patch (the one officially applied for 8.1+) to still generate pretty heavy queries, and for individual queries to continue to fail ... albeit without crashing your entire index :slightly_smiling_face:. I'm curious what your experience has been.
FWIW, we're running a different patch, which basically reverts https://issues.apache.org/jira/browse/LUCENE-8531, avoiding the exponential query expansion issue altogether (at the expense of somewhat inconsistent proximity query scoring, I guess?)"

### LINKED ISSUE #191: Debug Friday's crash
Use queries posted to #solr and investigate WordDelimiterGraphFilterFactory

### LINKED ISSUE #201: Add deploy script for new solrcloud (solr 8) staging environment
It will use the same zookeeper boxes, but with the new chroot, e.g.:

```
def zk_host
  "lib-zk-staging1:2181,lib-zk-staging2:2181,lib-zk-staging3:2181/solr8"
end
```

### PR DESCRIPTION
**Pull request by tpendragon**
fixes #187, fixes #188, fixes #191, fixes #201

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -12,10 +12,22 @@ solr_home_path = 'solr_configs'
...
+    version: '8.1.0',
     instance_dir: solr_instance_path,
     download_dir: solr_download_path,
     solr_options: {'s' => solr_home_path}
 }
 
 require 'solr_wrapper/rake_task'
+
+namespace :solr do
+  task :test_wait do
```
Comment (hackmastera) [inline Rakefile]: Perhaps we should delete this if we can't get it to create the cores. I looked into it a bit; seems like it must be related to that `solr_options: {'s' => solr_home_path}` but I couldn't figure out how that works. 

There is a `rake solr:stop` task that works for me.

File: config/deploy/production.rb [hunk 2]
```diff
@@ -11,13 +11,11 @@
 set :branch, ENV['BRANCH'] || 'master'
 
 def zk_host
-  "lib-zk1:2181,lib-zk2:2181,lib-zk3:2181"
+  "lib-zk1:2181,lib-zk2:2181,lib-zk3:2181/solr7"
```
Comment (jrgriffiniii) [inline config/deploy/production.rb]: Is this using the group membership feature of Zookeeper?

Comment (hackmastera) [reply config/deploy/production.rb]: This is what zookeeper calls "chroot" and what solr calls a "znode"

Comment (hackmastera) [reply config/deploy/production.rb]: This is what zookeeper calls "chroot" and what solr calls "znode"

**Comment by hackmastera** [comment]
thanks for looking @jrgriffiniii

**Pull request merged**